tests/test_groupsio.py

```diff
@@ -0,0 +1,391 @@
+        self.assertEqual(message['updated_on'], 1515301791.0)
+        self.assertEqual(message['category'], 'message')
+        self.assertEqual(message['tag'], 'https://groups.io/g/acme_group')
+
+        message = messages[45]
+        self.assertEqual(message['data']['Message-ID'], '<1528518785616723298.15102@groups.io>')
+        self.assertEqual(message['origin'], 'https://groups.io/g/acme_group')
+        self.assertEqual(message['uuid'], 'c1ab131763deb7b417e7f26cedb9ae17656ede15')
+        self.assertEqual(message['updated_on'], 1528518785.0)
+        self.assertEqual(message['category'], 'message')
+        self.assertEqual(message['tag'], 'https://groups.io/g/acme_group')
+
+    @httpretty.activate
+    def test_fetch_from_date(self):
+        """Test whether it fetches and parses messages since the given date"""
+
+        groupsio_mbox_archive = read_file('data/groupsio/messages.zip')
+
+        httpretty.register_uri(httpretty.GET,
+                               GROUPSIO_API_URL + 'downloadarchives',
+                               body=groupsio_mbox_archive)
+
+        from_date = datetime.datetime(2018, 5, 5)
+
+        backend = Groupsio('acme_group', self.tmp_path, 'aaaaa')
+        messages = [m for m in backend.fetch(from_date=from_date)]
+
+        self.assertEqual(len(messages), 8)
+
+        message = messages[0]
+        self.assertEqual(message['data']['Message-ID'], '<1526087603011004609.30544@groups.io>')
+        self.assertEqual(message['origin'], 'https://groups.io/g/acme_group')
+        self.assertEqual(message['uuid'], '078f740e54131b8833f69d605eae637b97797295')
+        self.assertEqual(message['updated_on'], 1526087603.0)
+        self.assertEqual(message['category'], 'message')
+        self.assertEqual(message['tag'], 'https://groups.io/g/acme_group')
+
+        message = messages[3]
+        self.assertEqual(message['data']['Message-ID'], '<1527915246656874452.2546@groups.io>')
+        self.assertEqual(message['origin'], 'https://groups.io/g/acme_group')
+        self.assertEqual(message['uuid'], 'c4a8db34d926f231ac99f5cc0eb340ed6b404e06')
+        self.assertEqual(message['updated_on'], 1527915246.0)
+        self.assertEqual(message['category'], 'message')
+        self.assertEqual(message['tag'], 'https://groups.io/g/acme_group')
+
+        message = messages[7]
+        self.assertEqual(message['data']['Message-ID'], '<1530334100637290090.7026@groups.io>')
+        self.assertEqual(message['origin'], 'https://groups.io/g/acme_group')
+        self.assertEqual(message['uuid'], '787835d4cc343299b8d1a427fa580365bb7feb0d')
+        self.assertEqual(message['updated_on'], 1530334100.0)
+        self.assertEqual(message['category'], 'message')
+        self.assertEqual(message['tag'], 'https://groups.io/g/acme_group')
+
+    @httpretty.activate
+    def test_fetch_empty(self):
+        """Test whether it works when a group does not store any mbox"""
```

Comment:
I think this documentation message is wrong because the code stores the mbox but the mbox is empty.